api/models/series.go

```diff
@@ -113,6 +114,16 @@ func (series SeriesByTarget) MarshalJSONFast(b []byte) ([]byte, error) {
 			// Replace trailing comma with a closing bracket
 			b[len(b)-1] = '}'
 		}
+		// 'stats' will not be included if request was proxied to graphite
+		b = append(b, `,"stats":{`...)
+		for stat, value := range s.Stats {
+			b = strconv.AppendQuoteToASCII(b, stat)
+			b = append(b, ':')
+			b = strconv.AppendFloat(b, value, 'f', -1, 64)
+			b = append(b, ',')
+		}
+		// Replace trailing comma with a closing bracket
+		b[len(b)-1] = '}'
```

Comment:
if for whatever reason `s.Stats` is empty then the resulting string will be broken json, because it would look something like `"stats":}`.  I think it would be better to add some check whether `s.Stats` is empty or not.

Reply:
I think this is the wrong place to do it.
we should track stats along with a request.
if we executed a request, we know the stats are valid.
it does mean we need to put some thought into how to return these stats without breaking any graphite clients which may not expect them.
since many api http endpoints respond with models.SeriesByTarget values, there doesn't seem to be a good place to put them, except for http response headers ?